pennylane/devices/qubit/preprocess.py

```diff
@@ -87,13 +96,30 @@ def expand_fn(circuit, max_expansion=10):
     if any(isinstance(o, MidMeasureMP) for o in circuit.operations):
         circuit = qml.defer_measurements(circuit)
 
-    if not all(_stopping_condition(op) for op in circuit.operations):
-        circuit = circuit.expand(depth=max_expansion, stop_at=_stopping_condition)
+    if len(circuit._prep) > 1:
+        raise DeviceError("DefaultQubit2 accepts at most one state prep operation.")
+
+    if not all(_accepted_operator(op) for op in circuit._ops):
+        new_ops = [final_op for op in circuit._ops for final_op in _operator_decomposition_gen(op)]
```

Comment:
I assume we're not worried about someday passing a recursion depth of 11 here? python itself would fail eventually...

Reply:
Not concerned about high recursion.  I could call getting rid of the "max depth" magic number a boon, rather than a problem. If we get into an infinite loop, Python itself will complain.

Othwise, I don't want unsupported operations to slip through because of a badly tuned magic number.

Reply:
I'm not generally a big fan of vague error messages. It's good to have a program break when it's supposed to rather than way later in the execution, but I'm wondering if it might be worth adding a try-except block here to catch a potential max recursion depth error and throw a more clear error.

Catching and re-throwing errors is unnecessary when the error is caused by user-facing functionality, but considering that the expansion is quite internal, I think this is an instance where it might be helpful.

Reply:
to be clear, I don't think we'll hit infinite recursion with regular flows. the only reason I can foresee is an unexpected bug (welcome to recursion). and it's hard to write error messages for bugs we can't expect! so i think we're all good here

Reply:
Now raising a `DeviceError` from `RecursionError` for a more informative error message. This would only really happen if the decomposition graph had an infinite loop.